traitlets/traitlets.py

```diff
@@ -908,6 +927,32 @@ def unobserve_all(self):
         """Remove all trait change handlers."""
         self._trait_notifiers = {}
 
+    def _register_validator(self, handler, name):
+        """Setup a handler to be called when a trait should be cross valdiated.
+
+        This is used to setup dynamic notifications for cross-validation.
+
+        Parameters
+        ----------
+        handler : callable
+            A callable that is called when the given trait is cross-validated.
+            Its signature is handler(proposal), where proposal is a dictionary
+            with the following keys:
+                - owner : the HasTraits instance
+                - value : the proposed value for the modified trait attribute
+                - name : the name of the modified trait attribute.
+        name : str
+            The name of the trait that should be cross-validated
+        """
+        if name in self._trait_validators:
+            raise TraitError("A cross-validator for the trait"
+                             " '%s' already exists" % name)
+        if hasattr(self, '_%s_validate' % name):
```

Comment:
same as observe above, check for ValidateHandler before issuing the deprecation warning.